contracts/peripheral/OhmBondManager.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.15;
+
+import {IBondSDA} from "../interfaces/IBondSDA.sol";
+import {IBondTeller} from "../interfaces/IBondTeller.sol";
+import {IEasyAuction} from "../interfaces/IEasyAuction.sol";
+import {IERC20} from "../interfaces/IERC20.sol";
+import {ITreasury} from "../interfaces/ITreasury.sol";
+import {IOlympusAuthority} from "../interfaces/IOlympusAuthority.sol";
+import {OlympusAccessControlled} from "../types/OlympusAccessControlled.sol";
+
+contract OhmBondManager is OlympusAccessControlled {
+    // ========= DATA STRUCTURES ========= //
+    struct BondProtocolParameters {
+        address callbackAddress;
+        uint256 initialPrice;
+        uint256 minPrice;
+        uint32 debtBuffer;
+        uint256 auctionTime;
+        uint32 depositInterval;
+    }
+
+    struct GnosisAuctionParameters {
+        uint256 auctionCancelTime;
+        uint256 auctionTime;
+        uint96 minRatioSold;
+        uint256 minBuyAmount;
+        uint256 minFundingThreshold;
+    }
+
+    // ========= STATE VARIABLES ========= //
+
+    /// Tokens
+    IERC20 public ohm;
+
+    /// Contract Dependencies
+    ITreasury public treasury;
+
+    /// Market Creation Systems
+    IBondSDA public fixedExpiryAuctioneer;
+    IBondTeller public fixedExpiryTeller;
+    IEasyAuction public gnosisEasyAuction;
+
+    /// Market parameters
+    BondProtocolParameters public bondProtocolParameters;
+    GnosisAuctionParameters public gnosisAuctionParameters;
+
+    constructor(
+        address ohm_,
+        address treasury_,
+        address feAuctioneer_,
+        address feTeller_,
+        address gnosisAuction_,
+        address authority_
+    ) OlympusAccessControlled(IOlympusAuthority(authority_)) {
+        ohm = IERC20(ohm_);
+        treasury = ITreasury(treasury_);
+        fixedExpiryAuctioneer = IBondSDA(feAuctioneer_);
+        fixedExpiryTeller = IBondTeller(feTeller_);
+        gnosisEasyAuction = IEasyAuction(gnosisAuction_);
+    }
+
+    // ========= MARKET CREATION ========= //
+    function createBondProtocolMarket(uint256 capacity_, uint256 bondTerm_) external onlyPolicy returns (uint256) {
+        _topUpOhm(capacity_);
+
+        /// Encodes the information needed for creating a bond market on Bond Protocol
+        bytes memory createMarketParams = abi.encode(
+            ohm, // payoutToken
+            ohm, // quoteToken
+            bondProtocolParameters.callbackAddress, // callbackAddress
```

Comment:
I would remove the callback option from this version. It doesn't make sense to use one in this context since the contract itself will be providing the OHM.